.github/workflows/maintainer_merge.yml

```diff
@@ -2,55 +2,37 @@ name: Maintainer merge
-  # the PR receives a review comment
-  pull_request_review_comment:
-    types: [created, edited]
+  repository_dispatch:
+    types: [maintainer-command]
 
 jobs:
   ping_zulip:
-    # we set some variables. The ones of the form `${{ X }}${{ Y }}` are typically not
-    # both set simultaneously: depending on the event that triggers the PR, usually only one is set
+    # we set some variables from the dispatched payload
     env:
-      AUTHOR: ${{ github.event.comment.user.login }}${{ github.event.review.user.login }}
-      PR_AUTHOR: ${{ github.event.issue.user.login }}${{ github.event.pull_request.user.login }}
-      PR_NUMBER: ${{ github.event.issue.number }}${{ github.event.pull_request.number }}
-      COMMENT_EVENT: ${{ github.event.comment.body }}
-      COMMENT_REVIEW: ${{ github.event.review.body }}
-      PR_TITLE_ISSUE: ${{ github.event.issue.title }}
-      PR_TITLE_PR: ${{ github.event.pull_request.title }}
-      PR_URL: ${{ github.event.issue.html_url }}${{ github.event.pull_request.html_url }}
-      EVENT_NAME: ${{ github.event_name }}
+      AUTHOR: ${{ github.event.client_payload.author }}
+      PR_AUTHOR: ${{ github.event.client_payload.pr_author }}
+      PR_NUMBER: ${{ github.event.client_payload.pr_number }}
+      COMMENT_EVENT: ${{ github.event.client_payload.comment_event }}
+      COMMENT_REVIEW: ${{ github.event.client_payload.comment_review }}
+      PR_TITLE_ISSUE: ${{ github.event.client_payload.pr_title_issue }}
+      PR_TITLE_PR: ${{ github.event.client_payload.pr_title_pr }}
+      PR_URL: ${{ github.event.client_payload.pr_url }}
+      EVENT_NAME: ${{ github.event.client_payload.event_name }}
     name: Ping maintainers on Zulip
     runs-on: ubuntu-latest
     if: github.repository == 'leanprover-community/mathlib4'
     steps:
-      - name: Find maintainer merge/delegate
+      - name: Set merge or delegate outputs from payload
         id: merge_or_delegate
         run: |
           echo "PR author: ${PR_AUTHOR}"
-          COMMENT="${COMMENT_EVENT}${COMMENT_REVIEW}"
 
-          # we strip `\r` since line endings from GitHub contain this character
-          COMMENT="${COMMENT//$'\r'/}"
-
-          # for debugging, we print some information
-          printf '%s' "${COMMENT}" | hexdump -cC
-          printf 'Comment:"%s"\n' "${COMMENT}"
-
-          m_or_d="$(printf '%s' "${COMMENT}" |
-            # captures `maintainer merge/delegate` as well as an optional `?`, ignoring subsequent spaces
-            sed -n 's=^maintainer  *\(merge\|delegate\)\(?\?\) *$=\1\2=p' | head -1)"
+          # Get values from the dispatched payload
+          m_or_d="${{ github.event.client_payload.mOrD }}"
 
           printf $'"maintainer delegate" or "maintainer merge" found? \'%s\'\n' "${m_or_d}"
 
-          printf $'mOrD=%s\n' "${m_or_d}" > "${GITHUB_OUTPUT}"
+          printf $'mOrD=%s\n' "${m_or_d}" >> "${GITHUB_OUTPUT}"
 
       - name: Check whether user is part of mathlib-reviewers team
```

Comment:
Likewise, we need to check the comment author from the payload. I think the user associated to the `repository_dispatch` event may not be who we expect.